### DISCUSSION TITLE
GET branch protection returning 404

### DISCUSSION THREAD

**Discussion by JoPintoPaul** [community/community#24582] [Apps, API and Webhooks]
<p>I’m trying to use the Github API to look at branch protection on a public repo.</p>
<p>I’m doing a GET request (via Postman but also tried via curl) to the following:<br>
<a href="https://api.github.com/repos/JoPintoPaul/xml-spike/branches/master/protection" rel="nofollow noopener">https://api.github.com/repos/JoPintoPaul/xml-spike/branches/master/protection</a></p>
<p>With the headers: </p>
<pre><code>Accept: application/vnd.github.luke-cage-preview+json
Accept:application/vnd.github.zzzax-preview+json
</code></pre>
<p>However, I keep getting the following 404 response:</p>
<pre><code>{
    "message": "Not Found",
    "documentation_url": "https://developer.github.com/v3/repos/branches/#get-branch-protection"
}
</code></pre>
<p>Anyone got any ideas?</p>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/jopintopaul">@jopintopaul</a> </p>
<p>Can you check whether the user is an admin for that repo?  Or is he in a team that has admin access to the repo? Because branch protection is something that is restricted.</p>
<p>In the below example you can see that I can able to get the master branch details but I cannot get the branch protection details since I am not the admin for that repo.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cc5ed61170b3c935945ddc32e0cfeaccf59f5bcf.png" data-download-href="/uploads/short-url/t9WAyqn8WYbjzdMbUmgrgpEieWX.png?dl=1" title="Screen Shot 2020-03-11 at 2.04.54 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085328-f79023fe-2567-483c-9584-619f39e8ac33.png" alt="Screen Shot 2020-03-11 at 2.04.54 PM.png" data-base62-sha1="t9WAyqn8WYbjzdMbUmgrgpEieWX" width="690" height="368" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cc5ed61170b3c935945ddc32e0cfeaccf59f5bcf_2_690x368.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/cc5ed61170b3c935945ddc32e0cfeaccf59f5bcf_2_1035x552.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/cc5ed61170b3c935945ddc32e0cfeaccf59f5bcf_2_1380x736.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cc5ed61170b3c935945ddc32e0cfeaccf59f5bcf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-03-11 at 2.04.54 PM.png</span><span class="informations">2828×1512 438 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Regards</p>
<p>Parithiban G</p>

**Comment by JoPintoPaul** [comment]
<p>Yes, my mistake, I didn’t realise that just viewing branch protection required authorized user. Issue resolved now.</p>
